lib/core/utils/xhr-q.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Returns a then(able) queue of XHR's
+ * @param {Object} config configuration for XMLHttpRequest
+ * @return {Object}
+ */
+axe.utils.xhrQ = (config) => {
```

Comment:
Writing your own XHR utility scares me. HTTP requests are complicated things and servers do not always behave in reasonable ways. I know there's not currently a way to `require()` modules in this code base, but we may want to consider putting a temporary solution to that problem in place so that we don't have to handle all of the weird edge cases that mature libraries like [Axios](https://github.com/axios/axios) and [Superagent](https://github.com/visionmedia/superagent) have already dealt with.

Maybe we could do something like this:

```
$ npm install axios
echo 'axe.utils.axios = (function () {' > lib/core/utils/axios.js
cat node_modules/axios/dist/axios.js >> lib/core/utils/axios.js
echo '  var axios = window.axios' >> lib/core/utils/axios.js
echo '  delete axios' >> lib/core/utils/axios.js
echo '  return axios' >> lib/core/utils/axios.js
echo '})()' >> lib/core/utils/axios.js
```

There's possibly even a Grunt plugin or something that could do this for us.

Reply:
See - #990